addOns/accessControl/src/main/java/org/zaproxy/zap/extension/accessControl/AccessControlAPI.java

```diff
@@ -0,0 +1,217 @@
+                    startOptions.raiseAlerts = params.getBoolean(PARAM_RAISE_ALERT);
+                } else {
+                    startOptions.raiseAlerts = true;
+                }
+
+                if (params.has(PARAM_ALERT_RISK_LEVEL)
+                        && StringUtils.isNotBlank(params.getString(PARAM_ALERT_RISK_LEVEL))) {
+                    startOptions.alertRiskLevel = params.getInt(PARAM_ALERT_RISK_LEVEL);
+                } else {
+                    startOptions.alertRiskLevel = org.parosproxy.paros.core.scanner.Alert.RISK_HIGH;
+                }
+
+                extension.startScan(startOptions);
+
+                result = ApiResponseElement.OK;
+                break;
+
+            case ACTION_WRITE_HTML_REPORT:
+                LOGGER.debug("Write HTML report called");
+
+                // The path for the XSL file
+                File xslFile =
+                        new File(
+                                Constant.getZapHome(),
+                                "xml" + File.separator + "reportAccessControl.html.xsl");
+
+                File reportFile = new File(params.getString(PARAM_FILENAME));
+
+                try {
+                    ReportGenerator.XMLToHtml(
+                            extension.generateLastScanXMLReport(params.getInt(PARAM_CONTEXT_ID)),
+                            xslFile.getAbsolutePath(),
+                            reportFile);
+                    result = new ApiResponseElement(name, "OK");
+                } catch (ParserConfigurationException e) {
+                    LOGGER.error("Failed to generate access control report:", e);
+                    result = new ApiResponseElement(name, e.getMessage());
+                }
+
+                // Have to add the check because ReportGenerator.XMLToHtml() won't raise an
+                // exception
+                if (reportFile.exists() == false || reportFile.canWrite() == false) {
+                    result =
+                            new ApiResponseElement(
+                                    name, "Error writing report to file " + reportFile.getPath());
+                }
+                break;
+            default:
+                throw new ApiException(ApiException.Type.BAD_ACTION);
+        }
+
+        return result;
+    }
+
+    @Override
+    public ApiResponse handleApiView(String name, JSONObject params) throws ApiException {
+        ApiResponse result;
+        String contextId = null;
+        switch (name) {
+            case VIEW_GET_SCAN_PROGRESS:
+                LOGGER.debug("Access control get scan progress called");
+
+                contextId = params.getString(PARAM_CONTEXT_ID);
+
+                result =
+                        new ApiResponseElement(
+                                name,
+                                String.valueOf(
+                                        extension.getScanProgress(Integer.parseInt(contextId))));
```

Comment:
This should catch `IllegalStateException` and return a better `ApiException` (`ApiException.Type.DOES_NOT_EXIST`?).